azure-pipelines.yml

```diff
@@ -0,0 +1,42 @@
+# Starter pipeline
+# Start with a minimal pipeline that you can customize to build and deploy your code.
+# Add steps that build, run tests, deploy, and more:
+# https://aka.ms/yaml
+
+trigger:
+  branches:
+    include:
+      - "*"
+
+pool:
+  vmImage: 'ubuntu-latest'
+
+container:
+  image: sonicdev-microsoft.azurecr.io:443/sonic-slave-buster:latest
+
+resources:
+  repositories:
+    - repository: SONiCSWSSCommon
+      type: github
+      name: Azure/sonic-swss-common
+      endpoint: Pterosaur
+
+steps:
+- checkout: self
+- checkout: SONiCSWSSCommon
+- script: |
+    sudo apt-get install -qq -y qtbase5-dev libdbus-glib-1-2 libdbus-glib-1-dev libpcsclite-dev docbook-to-man docbook-utils aspell-en libhiredis-dev libnl-3-dev libnl-genl-3-dev libnl-route-3-dev libnl-nf-3-dev swig3.0 libpython2.7-dev
+  displayName: "Install Dependencies"
+- script: |
+    ./autogen.sh
+    dpkg-buildpackage -us -uc -b
+  workingDirectory: sonic-swss-common
+  displayName: "Compile SONiC SWSS Common"
+- script: |
+    sudo dpkg -i libswsscommon_1.0.0_amd64.deb
+    sudo dpkg -i libswsscommon-dev_1.0.0_amd64.deb
+  displayName: "Install SONiC SWSS Common"
+- script: |
+    dpkg-buildpackage -us -uc -b
```

Comment:
is there way to archive the build artifacts? does this support pr build check?

Reply:
Yes, AzurePipeline support PR build check and build artifacts.
But I only know how to set PR build check but don't know how to provide artifacts for downloading.

Reply:
okay, it is a little bit bigger topic for artifacts, let's do that later.